app_test.go

```diff
@@ -258,6 +262,75 @@ func TestNewApp(t *testing.T) {
 	})
 }
 
+type eventLogger struct {
+	l *zap.Logger
+}
+
+func (l *eventLogger) LogEvent(event fxevent.Event) {
+	l.l.Info("emitted event", zap.String("event", reflect.TypeOf(event).String()))
+}
+
+func setupFxLogger(l *zap.Logger) fxevent.Logger {
+	return &eventLogger{l: l}
+}
+
+func TestSetupLogger(t *testing.T) {
+	t.Parallel()
+	t.Run("initializing custom logger", func(t *testing.T) {
+		t.Parallel()
+		app := fxtest.New(t,
+			Provide(zap.NewExample),
```

Comment:
This will write to stdout. Maybe we should use a zaptest.NewLogger here so it only writes to the testing.T.

```suggestion
			Supply(zaptest.NewLogger(t)),
```